Tries to parse non-pydantic metadata from `Annotated` as model data (and fails)

### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

When adding metadata to `Annotated` that pydantic does not understand, I expect it to ignore it. It seems, in the example, pydantic tries to parse the incoming data based on the `Metadata` model.

Here is the error:

```
tests/test_.py:13 (test_)
def test_():
        data = {"state": 2}
>       assert Model.model_validate(data)
E       pydantic_core._pydantic_core.ValidationError: 1 validation error for Model
E       state
E         Input should be a valid dictionary or instance of Metadata [type=model_type, input_value=2, input_type=int]
E           For further information visit https://errors.pydantic.dev/2.4/v/model_type

test_.py:16: ValidationError
```

PEP 593 states:

> "If a library (or tool) encounters a typehint Annotated[T, x] and has no special logic for metadata x, it should ignore it and simply treat the type as T.|

I do not see this happening here.

### Example Code

```Python
from typing import Annotated

from pydantic import BaseModel


class Metadata(BaseModel):
    ...


class Model(BaseModel):
    state: Annotated[int, Metadata()]


def test_():
    data = {"state": 2}
    assert Model.model_validate(data)
```


### Python, Pydantic & OS Version

```Text
pydantic version: 2.4.2
        pydantic-core version: 2.10.1
          pydantic-core build: profile=release pgo=false
                 install path: /Users/cleancoder/Library/Caches/pypoetry/virtualenvs/xxx-sT27uenX-py3.9/lib/python3.9/site-packages/pydantic
               python version: 3.9.15 (main, Feb 20 2023, 14:56:22)  [Clang 14.0.0 (clang-1400.0.29.202)]
                     platform: macOS-13.4.1-arm64-arm-64bit
             related packages: email-validator-2.0.0.post2 pydantic-extra-types-2.1.0 typing_extensions-4.7.1 pydantic-settings-2.0.3 fastapi-0.103.2
```

--- a/docs/concepts/json_schema.md
+++ b/docs/concepts/json_schema.md
@@ -524,6 +524,52 @@ except ValidationError as e:
     """
 ```
 
+As seen above, annotating a field with a `BaseModel` type can be used to modify or override the generated json schema.
+However, if you want to take advantage of storing metadata via `Annotated`, but you don't want to override the generated JSON
+schema, you can use the following approach with a no-op version of `__get_pydantic_core_schema__` implemented on the
+metadata class:
+
+```py
+from typing import Type
+
+from pydantic_core import CoreSchema
+from typing_extensions import Annotated
+
+from pydantic import BaseModel, GetCoreSchemaHandler
+
+
+class Metadata(BaseModel):
+    foo: str = 'metadata!'
+    bar: int = 100
+
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls, source_type: Type[BaseModel], handler: GetCoreSchemaHandler
+    ) -> CoreSchema:
+        if cls is not source_type:
+            return handler(source_type)
+        return super().__get_pydantic_core_schema__(source_type, handler)
+
+
+class Model(BaseModel):
+    state: Annotated[int, Metadata()]
+
+
+m = Model.model_validate({'state': 2})
+print(repr(m))
+#> Model(state=2)
+print(m.model_fields)
+"""
+{
+    'state': FieldInfo(
+        annotation=int,
+        required=True,
+        metadata=[Metadata(foo='metadata!', bar=100)],
+    )
+}
+"""
+```
+
 ## JSON schema types
 
 Types, custom field types, and constraints (like `max_length`) are mapped to the corresponding spec formats in the

--- a/docs/errors/usage_errors.md
+++ b/docs/errors/usage_errors.md
@@ -38,14 +38,19 @@ class Foo(BaseModel):
     a: Optional['Bar'] = None
 
 
+try:
+    # this doesn't work, see raised error
+    foo = Foo(a={'b': {'a': None}})
+except PydanticUserError as exc_info:
+    assert exc_info.code == 'class-not-fully-defined'
+
+
 class Bar(BaseModel):
     b: 'Foo'
 
 
-try:
-    foo = Foo(a={'b': {'a': None}})
-except PydanticUserError as exc_info:
-    assert exc_info.code == 'class-not-fully-defined'
+# this works, though
+foo = Foo(a={'b': {'a': None}})
 ```
 
 For BaseModel subclasses, it can be fixed by defining the type and then calling `.model_rebuild()`:

--- a/pydantic/type_adapter.py
+++ b/pydantic/type_adapter.py
@@ -219,9 +219,14 @@ class TypeAdapter(Generic[T]):
         """Initializes the TypeAdapter object.
 
         Args:
-            type: The Python type associated with the `TypeAdapter`.
+            type: The type associated with the `TypeAdapter`.
             config: Configuration for the `TypeAdapter`, should be a dictionary conforming to [`ConfigDict`][pydantic.config.ConfigDict].
             _parent_depth: depth at which to search the parent namespace to construct the local namespace.
+
+        !!! note
+            You cannot use the `config` argument when instantiating a `TypeAdapter` if the type you're using has its own
+            config that cannot be overridden (ex: `BaseModel`, `TypedDict`, and `dataclass`). A
+            [`type-adapter-config-unused`](../errors/usage_errors.md#type-adapter-config-unused) error will be raised in this case.
 
         !!! note
             The `_parent_depth` argument is named with an underscore to suggest its private nature and discourage use.

--- a/pydantic/type_adapter.py
+++ b/pydantic/type_adapter.py
@@ -216,7 +216,21 @@ class TypeAdapter(Generic[T]):
             ...
 
     def __init__(self, type: Any, *, config: ConfigDict | None = None, _parent_depth: int = 2) -> None:
-        """Initializes the TypeAdapter object."""
+        """Initializes the TypeAdapter object.
+
+        Args:
+            type: The Python type associated with the `TypeAdapter`.
+            config: Configuration for the `TypeAdapter`, should be a dictionary conforming to [`ConfigDict`][pydantic.config.ConfigDict].
+            _parent_depth: depth at which to search the parent namespace to construct the local namespace.
+
+        !!! note
+            The `_parent_depth` argument is named with an underscore to suggest its private nature and discourage use.
+            It may be deprecated in a minor version, so we only recommend using it if you're
+            comfortable with potential change in behavior / support.
+
+        Returns:
+            A type adapter configured for the specified `type`.
+        """
         config_wrapper = _config.ConfigWrapper(config)
 
         try:

